### PR TITLE
Expose styles as variables

### LINKED ISSUE #565: Expose styles as variables?
A package I am maintaining [chalk-pipe](https://github.com/LitoMore/chalk-pipe) has a function to determine what type of style is passed in.

Currently Chalk only exposes the types of styles. But in my case, I want the variables.

https://github.com/LitoMore/chalk-pipe/blob/main/source/index.ts#L3-L23

Could we add some exports like this:

```js
export const modifiers = ['rest', 'bold' , 'dim', ...];
export const colors = = ['black', 'red', 'green', ...];
```

Since we already documented which styles are supported in the readme. It's a good idea to expose these styles as variables.

### PR DESCRIPTION
**Pull request by LitoMore**
Resolves #565

### CODE REVIEW

File: source/index.js [hunk 1]
```diff
@@ -209,4 +209,8 @@ export {
 	stderrColor as supportsColorStderr,
 };
 
+export const modifiers = Object.keys(ansiStyles.modifier);
+export const foregroundColors = Object.keys(ansiStyles.color);
+export const backgroundColors = Object.keys(ansiStyles.bgColor);
```
Comment (LitoMore) [inline source/index.js:214]: This makes our Chalk slower. Is this acceptable?

Comment (Qix-) [reply source/index.js:214]: That's very strange. Can you show how?

Comment (LitoMore) [reply source/index.js:214]: This makes the code slightly slower, but it might be negligible.

Comment (sindresorhus) [inline source/index.js:214]: Need to export `colors` too. Combination of foreground and background.

**Comment by sindresorhus** [comment]
This needs to be documented.

File: source/index.d.ts [hunk 2]
```diff
@@ -284,4 +284,8 @@ export {
 // } from '#supports-color';
 } from './vendor/supports-color/index.js';
 
+export const modifiers: Modifiers[];
```
Comment (sindresorhus) [inline source/index.d.ts:287]:
```suggestion
export const modifiers: readonly Modifiers[];
```

File: readme.md [hunk 3]
```diff
@@ -201,6 +201,10 @@ Explicit 256/Truecolor mode can be enabled using the `--color=256` and `--color=
 
 `chalkStderr` contains a separate instance configured with color support detected for `stderr` stream instead of `stdout`. Override rules from `supportsColor` apply to this too. `supportsColorStderr` is exposed for convenience.
 
+### modifiers, foregroundColors, backgroundColors, and colors
+
+All supported style strings are exposed for convenience. `colors` is the combination of `foregroundColors` and `backgroundColors`.
+
```
Comment (sindresorhus) [inline readme.md:207]: I think an example could be useful, as it's not clear what these properties contain or what to use them for.

**Pull request merged**